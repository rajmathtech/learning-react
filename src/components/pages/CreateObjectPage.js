import React from 'react';
import {connect} from 'react-redux';
import ObjectForm from '../ObjectForm';
import { addObject } from '../../redux/actions/object';
export class CreateObjectPage extends React.Component{
    onSubmitForm= (object) => {
        this.props.addObject(object);
        this.props.history.push('/'); 
     };
    render () {
        return (<div className="container__main">
        <h1 className="heading-primary"> Welcome to Create Object</h1>
        <ObjectForm  object={{}} onSubmitForm={this.onSubmitForm}/>
        </div>)
    }
} 

const mapDispatchToProps = (dispatch) => ({
    addObject:(object)=>dispatch(addObject(object))
});

export default connect(undefined, mapDispatchToProps)(CreateObjectPage);