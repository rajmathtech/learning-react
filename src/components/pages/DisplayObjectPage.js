import React from 'react';
import { Link } from 'react-router-dom';
import { deleteObject } from '../../redux/actions/object';
import { connect } from 'react-redux';
const DisplayObjectPage = (props) => {
    let obj=undefined;const id=props.match.params.id;
    props.objects.filter((object) => 
    {
        if (object.id === props.match.params.id) {
            obj=object;
            return true;
        }
    });
    const onDeleteObject = () => {
        props.dispatch(deleteObject(obj));  
        props.history.push('/');  
    }
    return (<div className="fullPage"> 
        <div className="popup-content">
            <h1> Welcome to Edit Object {id}</h1>
            {obj.details}
            <Link to="/" className="editLink"> Go back</Link>
            <Link to={`/edit/${id}`} className="editLink"> Edit</Link>
            <button onClick={onDeleteObject}> Delete </button>
        </div>  
    </div>);
};
    
const mapStateToProps = (state)=>
{ 
    return {
        objects:state.objects
    };
};
export default connect(mapStateToProps)(DisplayObjectPage);
