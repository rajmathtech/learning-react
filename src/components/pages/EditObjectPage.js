import React from 'react';
import ObjectForm from '../ObjectForm';
import {connect} from 'react-redux';
import { editObject }  from '../../redux/actions/object';
export class EditObjectPage extends React.Component{
    onSubmitForm = (update) => {    
        this.props.editObject(this.props.object, update);
        this.props.history.push('/');
    };
    render () {
        
    return (<div className="container__main">
    <h1 className="heading-primary"> Welcome to Edit Object</h1>
    <ObjectForm object={this.props.object} onSubmitForm={this.onSubmitForm}/>
    </div>);
    }
}
const mapDispatchToProps = (dispatch) => ({
    editObject: (obj, object) => dispatch(editObject(obj, object))
})
 const mapStateToProps = (state, props) =>({
    object:state.objects.find((object) => object.id === props.match.params.id)
 });

export default connect(mapStateToProps, mapDispatchToProps)(EditObjectPage);