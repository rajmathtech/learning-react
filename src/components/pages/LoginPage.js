import React from 'react';
import {connect} from 'react-redux';
import {startLogin} from '../../redux/actions/auth';
export const LoginPage = ({startLogin}) => (
    <div> 
    <h1> welcome to Log in Page </h1>
    <button onClick={startLogin}> LogIn </button>
    </div>
);
const mapDispatchToProps = (dispatch) =>{
    return {
        startLogin: () => dispatch(startLogin())
    }
}
export default connect(undefined, mapDispatchToProps)(LoginPage);