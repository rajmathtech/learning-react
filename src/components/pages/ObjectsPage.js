import React from 'react';
import Objects from '../Objects';
const ObjectsPage = () => (<div className="container__main"> 
    <h1 className="heading-primary"> Welcome to Objects list </h1> 
    <Objects />
</div>);

export default ObjectsPage;