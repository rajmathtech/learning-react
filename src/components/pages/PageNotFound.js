import React from 'react';

const PageNotFound = () => (<div> <h2> 404- page not found! </h2> </div>);

export default PageNotFound;