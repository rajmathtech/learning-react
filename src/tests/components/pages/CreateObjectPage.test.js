import React from "react";
import { shallow } from "enzyme";
import { CreateObjectPage } from "../../../components/pages/CreateObjectPage";
import objects from "../../fixtures/objects";
let addObjectSpy, historySpy, wrapper;
beforeEach(() => {
  addObjectSpy = jest.fn();
  historySpy = { push: jest.fn() };
  wrapper = shallow(
    <CreateObjectPage startAddObject={addObjectSpy} history={historySpy} />
  );
});
test("shoukd render create object page component", () => {
  expect(wrapper).toMatchSnapshot();
});
test("shoukd render create object page component with object one", () => {
  wrapper.find("ObjectForm").prop("onSubmitForm")(objects[0]);
  expect(addObjectSpy).toHaveBeenLastCalledWith(objects[0]);
  expect(historySpy.push).toHaveBeenLastCalledWith("/");
});
