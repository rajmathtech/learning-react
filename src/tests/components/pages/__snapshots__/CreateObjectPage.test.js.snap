// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shoukd render create object page component 1`] = `
<div
  className="container__main"
>
  <h1
    className="heading-primary"
  >
     Welcome to Create Object
  </h1>
  <ObjectForm
    object={Object {}}
    onSubmitForm={[Function]}
  />
</div>
`;
